### PR TITLE
scrollWindow: fix view scrolling

### PR DESCRIPTION
**Pull request by meven**
Change-Id: I54990645d17e1e352397f1d57b4cc671af6b5839

* Resolves: #9770
* Target version: master 

### Summary

 * make scrollWindow height fix
 * special case data.horizontal.page_size == 0
 * improve computation of rowHeight
 * don't resize dynamically dimensions based on dom changes

### Checklist

- [x] I have run `make prettier-write` and formatted the code.
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/control/jsdialog/Widget.ScrolledWindow.js [hunk 1]
```diff
@@ -54,7 +54,8 @@
 		scrollwindow.id = data.id;
 
 	// drawing areas inside scrollwindows should be not cropped so we add special class
-	if (_hasDrawingAreaInside(data.children))
+	let drawingArea = _hasDrawingAreaInside(data.children)
```
[the remark this answers is not in the document]
Comment (caolanm) [reply browser/src/control/jsdialog/Widget.ScrolledWindow.js:57]: The auto-bot has a good point here about a missing semicolon on this line

**Comment by eszkadev** [comment]
It works nice in most of the cases, but still has problems with vertical scrolling in "managed by user" mode.

[symbols-2024-08-22_10.33.22.webm](https://github.com/user-attachments/assets/f21ea35a-3b81-4c11-98a6-5926c0f7e5c1)

It worked in the past before regression commit, it was also not doing "scrolling animation" so UI looked less buggy/jumping.

**Review by eszkadev — CHANGES_REQUESTED** [review]
please try to fix vertical scrolling

**Comment by meven** [comment]
The use cases to test:

In calc:
 * function wizard, select AGGREGATE and try to scroll
 * insert symbol
 * properties > custom properties tab.

**Comment by eszkadev** [comment]
This works bad:
[scroll-2024-08-30_08.54.08.webm](https://github.com/user-attachments/assets/6b567266-56e3-4d9f-8768-2edc2fed5c89)

I used keyboard (arrow down) in symbol dialog.

Please see how it works with the reverted version (current master):
[scroll-2024-08-30_08.57.03.webm](https://github.com/user-attachments/assets/931c3a9b-fa58-4eef-a589-a94b48cb1dd0)

Can we just setup the size of scroll window in the file properties dialog for custom properties using CSS?
I think behavior now is ok, only look of custom properties tab is not good.

**Comment by meven** [comment]
> Can we just setup the size of scroll window in the file properties dialog for custom properties using CSS?

The content size is dynamic in the custom properties and function wizard, there it is not possible and in the symbol dialog it would break when we add new symbols.

**Comment by meven** [comment]
> This works bad: [scroll-2024-08-30_08.54.08.webm](https://github.com/user-attachments/assets/6b567266-56e3-4d9f-8768-2edc2fed5c89)
> 
> I used keyboard (arrow down) in symbol dialog.

I noticed it, I have an idea how to fix that.

**Comment by meven** [comment]
> Really isn't it enough to fix custom properties sizing on top of current master ?
> Like some CSS: to make scroll-window height depending on height of the dialog and rest of the items static.

My patches sets the scroll-window "view" height, in `browser/css/jsdialogs.css`.
But that's the content-height that's dynamic.

**Comment by eszkadev** [comment]
not needed currently: other PR was merged

**Pull request closed without merging**